recipes/qarchive/all/conanfile.py

```diff
@@ -29,14 +29,11 @@ class QarchiveConan(ConanFile):
         "fPIC": True,
     }
 
-    generators = "cmake", "cmake_find_package"
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _qt_major(self):
+        return Version(self.deps_cpp_info["qt"].version).major
```

Comment:
```suggestion
        return Version(self.dependencies["qt"].ref.version).major
```

https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-generate-method
https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface